### PR TITLE
Feature: Add checks for load_model() and default args.

### PR DESCRIPTION
**Pull request by tiaguinho-code**
Adding some checks to make the load_models function easier to use and give more meaningful errors for when it's used from the cli. 
Added a default arg for the random forrest and prepared code for other model's default args.

### CODE REVIEW

File: nmrcraft/models/models.py [hunk 1]
```diff
@@ -1,8 +1,42 @@
...
+
+
+def validate_kwargs(kwargs, model_class, model_name):
+    args = inspect.signature(model_class.__init__).parameters.keys()
+    for kwarg in kwargs:
+        if kwarg not in args:
+            raise InvalidArgumentError(kwarg, model_name)
+
+
 def load_model(model_name, **kwargs):
```
Comment (mlederbauer) [inline nmrcraft/models/models.py]: Let’s add a trype hint here, which makes the “InvalidModelNameError” obsolete. By adding a type hint, you atomatically force the model_name to be a string:

def load_model(model_name:str, **kwargs):

File: nmrcraft/models/models.py [hunk 2]
```diff
@@ -1,8 +1,42 @@
+import inspect
+
 from sklearn.ensemble import GradientBoostingClassifier, RandomForestClassifier
 from sklearn.linear_model import LogisticRegression
 from sklearn.svm import SVC
 
 
+class InvalidModelNameError(ValueError):
```
Comment (mlederbauer) [inline nmrcraft/models/models.py:9]: see comment below at line 40

File: nmrcraft/models/models.py [hunk 3]
```diff
@@ -21,6 +55,30 @@ def load_model(model_name, **kwargs):
...
+    try:
+        validate_model_instance(model_name)
+        model_class = models.get(model_name.lower())
+        validate_model_availability(model_name, models)
+    except Exception as e:
+        print(f"Error loading model: {e}")
+        return None
+
+    if model_name == "random_forest":
+        kwargs.setdefault("n_jobs", -1)  # Set max number of jobs
```
Comment (mlederbauer) [inline nmrcraft/models/models.py:64]: what is the n_jobs for?

File: nmrcraft/models/models.py [hunk 4]
```diff
@@ -21,6 +55,30 @@ def load_model(model_name, **kwargs):
...
+
+    # if(model_name == "gradient_boosting"):
+
+    # if(model_name == "logistic_regression"):
+
+    # if(model_name == "svc"):
```
Comment (mlederbauer) [inline nmrcraft/models/models.py]: Do we need those if statements for now ?

**Review by mlederbauer — COMMENTED** [review]
Adding the checks is very much appreciated to be able to have a clean code and no surprising errors! I have an idea how we can streamline this more and will add a few comments shortly!

**Comment by mlederbauer** [comment]
Hi! I rewrote the code a bit to use the “custom errors” you created in sequence, e.g. first checking if the model is available, secondly checking is the passed kwargs are correct. What do you think?

**Comment by tiaguinho-code** [comment]
> Hi! I rewrote the code a bit to use the “custom errors” you created in sequence, e.g. first checking if the model is available, secondly checking is the passed kwargs are correct. What do you think?

I like this. This looks less complex then that nested function stuff I got bullied into by ruff. Very straightforward and easy to read.

**Comment by mlederbauer** [comment]
Ah sorry to hear about this! In the end, it helps me to collect my thoughts and think “step by step” for how to “un-couple” or uncomplicate code, and as a last check also check if there is anything that is redundant. It also helps to ask ChatGPT for feedback on “how to refactor the existing code”. Hope ruff won’t cause too annoying problems in the future, otherwise let us know!

**Review by mlederbauer — APPROVED** [review]

**Pull request merged**